src/ui/EditorNS/editor.cpp

```diff
@@ -141,37 +141,33 @@ namespace EditorNS
         if (msg.startsWith("[ASYNC_REPLY]")) {
             QRegExp rgx("\\[ID=(\\d+)\\]$");
 
-            if(rgx.indexIn(msg) == -1)
-                return;
-
-            if (rgx.captureCount() != 1)
+            if (rgx.indexIn(msg) == -1 || rgx.captureCount() != 1)
                 return;
 
             unsigned int id = rgx.capturedTexts()[1].toInt();
 
             // Look into the list of callbacks
-            for (auto it = this->asyncReplies.begin(); it != this->asyncReplies.end(); ++it) {
-                if (it->id == id) {
-                    auto cb = it->callback;
-                    it->value->set_value(data);
-                    this->asyncReplies.erase(it);
-
-                    if (cb != 0) {
```

Comment:
Change you change that to `cb != nullptr`? It quenches clang warnings about using zero as nullpointer constant.